disco_aws_automation/disco_deploy.py

```diff
@@ -390,6 +391,103 @@ def handle_tested_ami(self, old_hostclass_dict, ami, desired_size,
         self._disco_aws.terminate(self._get_new_instances(ami.id), use_autoscaling=True)
         self._disco_aws.spinup([post_hostclass_dict])
 
+    # Disable too many local variables, branches, and statements because this method handles blue/green from
+    # end to end.
+    # pylint: disable=too-many-locals,too-many-branches,too-many-statements
+    def handle_blue_green_ami(self, pipeline_dict, ami, old_group,
+                              deployable=False, run_tests=False, dry_run=False):
+        '''
+        Tests hostclasses which we can deploy normally
+
+        Deploys AMIs into a new autoscaling group. If the new AMI passes tests, the old ASG is destroyed,
+        and the new ASG is preserved. Otherwise, the original ASG is preserved.
+
+        Also creates a separate testing ELB that is used for the purposes of integration tests.
+        '''
+        logging.info("testing %s hostclass %s AMI %s with %s deployment strategy",
+                     "deployable" if deployable else "non-deployable", pipeline_dict["hostclass"], ami.id,
+                     DEPLOYMENT_STRATEGY_BLUE_GREEN)
+
+        hostclass = pipeline_dict["hostclass"]
+        uses_elb = is_truthy(self.hostclass_option_default(hostclass, "elb", "no"))
+
+        if dry_run:
+            return
+
+        new_group_config = copy.deepcopy(pipeline_dict)
+        new_group_config["sequence"] = 1
+        new_group_config["smoke_test"] = "no"
+        new_group_config["ami"] = ami.id
+
+        # If there is an already existing ASG, use its sizing. Otherwise, use the pipeline's sizing.
```

Comment:
Talking to Pratap, it's probably more important to use the existing group's sizing rather than the pipeline, because we might manually resize a group outside of its pipeline, or alarms might resize a group. So I'll leave this the way it is.